src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -227,7 +236,7 @@ public class ObjectExecutionNode : ExecutionNode, IParentExecutionNode
         /// <summary>
         /// Returns a dictionary of child execution nodes, with keys set to the names of the child fields that the child nodes represent.
         /// </summary>
-        public IDictionary<string, ExecutionNode> SubFields { get; set; }
+        public Dictionary<string, ExecutionNode> SubFields { get; set; }
```

Comment:
There was **A LOT** of allocations here due to repeatedly traversed dictionaries and `Dictionary<>+Iterator` boxing.